sqladmin/forms.py

```diff
@@ -161,107 +199,169 @@ async def convert(
                     ]
                     kwargs["object_list"] = object_list
 
+        return kwargs
+
+    async def convert(
+        self,
+        model: type,
+        mapper: Mapper,
+        prop: Union[ColumnProperty, RelationshipProperty],
+        engine: Union[Engine, AsyncEngine],
+        field_args: Dict[str, Any] = None,
+        label: Optional[str] = None,
+        override: Optional[Type[Field]] = None,
+    ) -> Optional[UnboundField]:
+
+        kwargs = await self._prepare_kwargs(
+            model=model,
+            mapper=mapper,
+            prop=prop,
+            engine=engine,
+            field_args=field_args,
+            label=label,
+            override=override,
+        )
+
+        if kwargs is None:
+            return None
+
         if override is not None:
             assert issubclass(override, Field)
             return override(**kwargs)
 
-        converter = self.get_converter(prop)
-
-        return converter(
-            model=model, mapper=mapper, prop=prop, column=column, field_args=kwargs
-        )
+        converter = self.get_callback(model=model, prop=prop)
+        return converter(model=model, prop=prop, kwargs=kwargs)
 
     def get_pk(self, o: Any, pk_name: str) -> Any:
         return getattr(o, pk_name)
 
 
 class ModelConverter(ModelConverterBase):
-    @classmethod
-    def _string_common(cls, column: Column, field_args: Dict, **kwargs: Any) -> None:
+    @staticmethod
+    def _string_common(prop: ColumnProperty) -> List[Validator]:
+        li = []
+        column: Column = prop.columns[0]
         if isinstance(column.type.length, int) and column.type.length:
-            field_args["validators"].append(validators.Length(max=column.type.length))
+            li.append(validators.Length(max=column.type.length))
+        return li
```

Comment:
Isn't this always a list with one element?

Reply:
The list can also be empty.

There are a few ways to write this snippet; I refactored because I thought it'd be simpler if the function did not rely on side effects.